### DISCUSSION TITLE
Not a registered object

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#28879] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello, I modified a postprocessor object and compiled it susccessfully. I used it yesterday but when I open my application to run the same script today I get this error.
```
A 'ElementIntegralVariablePostprocessor_new' is not a registered object.check_inp(moose_srv)
If you are trying to find this object in a dynamically loaded library, make sure thatcheck_inp(moose_srv)
the library can be found either in your "Problem/library_path" parameter or in thecheck_inp(moose_srv)
MOOSE_LIBRARY_PATH environment variable.
```
I even changed the name of the object and compiled it successfully. It did not work.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Use `MooseApp` for the app to register the object to, then recompile

**Reply by SY1104** [reply]
Hello!could you please describe how to do this in detai?

**Reply by GiudGiud** [reply]
at the top of any source file, there is a call to 
```
registerMooseObject("MiscApp", ThermoDiffusion);
```
the first one is the app name, the second the object name.
The app name has to be exact. Any typo or capitalization error, and the code wont recognize where the object is coming from
